lib/_stream_writable.js

```diff
@@ -53,6 +53,13 @@ Object.setPrototypeOf(Writable, Stream);
 
 function nop() {}
 
+function writableErrorOrDestroy (stream, er) {
+  // Some modules unfortunately override `stream.destroy()` which otherwise
+  // performs this logic.
+  stream._writableState.errorEmitted = Boolean(er);
```

Comment:
@mcollina: actually this confuses me, if `destroy` is **not** override then this will actually cause the error not to be emitted and if it is override then `errorEmitted` isn't really relevant anymore?